### PR TITLE
feat: preview Nomad beta, add tutorials

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Spikes out a rollout of Nomad content, and marks Nomad as "beta" _but only in `preview`_, not in `production`.

## 🤷 Why

Nomad is one of the many products we plan to roll out to the HashiCorp Developer beta.

## 🛠️ How

- Adds `nomad` to our array of products marked as `"beta"`
- Adds `/nomad/tutorials` routes

## 🧪 Testing

Visit [the preview][preview], click around the Nomad content, everything should look and function as expected. Specifically

- Nomad should appear in the "beta" in preview links
    - on [the home page][preview], and in nav menus, the Nomad product items should be colourful, not grayed out
- Nomad docs routes should look and function as expected
    - [/nomad][/nomad] should render a product landing page with placeholder content
    - [/nomad/docs][/nomad/docs] should work
        - 🚧 Note: custom `/docs` landing page _intentionally_ not implemented yet. See "Anything else" section for details.
        - Further routes should render docs like https://www.nomadproject.io/docs
        - 🔎 Search should function as it does on https://www.nomadproject.io/docs
    - [/nomad/api-docs][/nomad/api-docs] should render docs like https://www.nomadproject.io/api-docs
    - [/nomad/plugins][/nomad/plugins] should render docs like https://www.nomadproject.io/plugins
    - [/nomad/tools][/nomad/tools] should render docs like https://www.nomadproject.io/tools
- Nomad tutorial routes should look and function as expected
    - [/nomad/tutorials][/nomad/tutorials] should render a tutorial landing page, with content identical to https://learn.hashicorp.com/nomad
    - Collections and Tutorials accessible from that landing page should look and function as expected
- Nomad install view should look & function as expected
    - [/nomad/downloads][/nomad/downloads] should resemble https://www.nomadproject.io/downloads

## 💭 Anything else?

Main intent here is to stub placeholder content in, and test the "beta in preview" flow. Intent is _not_ to have Nomad "dev complete".

- ➡️ This PR is based on #708, hoping to merge #708 first
- 🚧 Content is placeholder for now
    - Collapsable sections below have some details
    - I'm working on further details in a Google Doc, at 📜  [2022-07-14 - Nomad in Dev Dot - Eng Spike Notes](https://docs.google.com/document/d/1B2hGQsGqbb6dmUYT9JJXk6JjnVWt0ncr4MNNpzgVsUk/edit#)
- 🚧 I have not yet stubbed in "custom" docs landing pages for [/nomad/docs][/nomad/docs], [/nomad/api-docs][/nomad/api-docs], [/nomad/plugins][/nomad/plugins], or [/nomad/tools][/nomad/tools]
     - Custom landing page at /nomad/docs should be analogous to what's at https://www.nomadproject.io/docs
    - These "custom" pages will likely be needed on _all_ docs category routes as we want to add a "hero section at the top of each page"
    - I've deferred this work for now, I think it might be valuable to spike out consolidating `/nomad/<docs-category>` under one `src/pages/nomad/[...allDocs].tsx` page file, similar to what we've done in the Terraform spike at #705 
    - See 🎟️ [✓ Scope Nomad](https://app.asana.com/0/1202520168081556/1202536208758638/f) for details)

## Partial notes on content gaps

I'm working on more thorough notes in  📜  [2022-07-14 - Nomad in Dev Dot - Eng Spike Notes](https://docs.google.com/document/d/1B2hGQsGqbb6dmUYT9JJXk6JjnVWt0ncr4MNNpzgVsUk/edit#). I'm retaining the sections below for posterity.

<details>

<summary>🚨 Major content gap on the <a href="https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad">/nomad</a> landing page</summary>

What's there now is very much placeholder, we'd definitely want to fix that up before actually rolling this out.

Screenshot 2022-07-08 at 10-35-54 Nomad HashiCorp Developer

</details>

<details>

<summary>⚠️ Minor content gap on the <a href="https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/downloads">/nomad/downloads</a> page</summary>

Content missing from sidecar-like layout, showing `Lorem ipsum` currently. Other products include a brief description.

At worst we can likely copy something in from another part of the Nomad site (for example, we could use `Nomad is a simple and flexible scheduler and orchestrator to deploy and manage containers and non-containerized applications across on-prem and clouds, at scale.`, from <https://www.nomadproject.io/>).

Screen Shot 2022-07-08 at 10 35 38

</details>

[task]: https://app.asana.com/0/1202354188341171/1202354188341389/f
[preview]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/
[/nomad]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad
[/nomad/docs]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/docs
[/nomad/api-docs]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/api-docs
[/nomad/plugins]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/plugins
[/nomad/tools]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/tools
[/nomad/tutorials]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/tutorials
[/nomad/downloads]: https://dev-portal-git-zsspike-nomad-rollout-hashicorp.vercel.app/nomad/downloads

### CODE REVIEW

File: config/base.json [hunk 1]
```diff
@@ -14,8 +14,7 @@
 		"canonical_base_url": "https://developer.hashicorp.com",
 		"max_static_paths": 10,
 		"revalidate": 360,
-		"content_preview_branch": "dev-portal",
-		"beta_product_slugs": ["waypoint", "vault"],
```
Comment (kendallstrautman) [inline config/base.json:18]: One idea that we could explore is just enabling waypoint as a beta slug in the 'preview' config. I was going to try this to enable the product for previews in the tutorials repo

**Comment by zchsh** [comment]
@ashleemboyer @kendallstrautman @BRKalow I've updated this Nomad "spike" PR, having done a bunch of config-reduction work in separate PRs since first opening things here.

Let me know what ya'll think 🙇‍♂️

---

For further context, the "config-reduction" PRs are:
- #691
- #697 
- #698 
- #699 
- #704 
- #708 (pending, hoping to get reviewed and merged before this PR!)

With less need for additional config as we onboard new products, this PR now contains only the following work:
- Mark Nomad as "beta" in `preview` config
- Make `dev` config extend `preview` (rather than `base`)
- Add `/nomad/tutorials` routes

---

**Note on "custom" docs views**: the intent here is to follow up this PR with another PR that adds in the "custom" docs landing pages. I'm hoping to use the `[...allDocs].tsx` page file approach for this, so felt right to split that work into a separate PR rather than try to tack it on here.

File: config/development.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"extends": "base",
+	"extends": "preview",
```
Comment (BRKalow) [inline config/development.json:2]: This doesn't feel right. If we're trying to get the same beta product slugs in preview and development, we should think about moving the expanded list to `base` and limiting the list in `production.json`

Comment (zchsh) [reply config/development.json:2]: Sweet, matching `preview` and `dev` was definitely the intent. I'll update to limit the list in `production.json` 👍

**Comment by ashleemboyer** [comment]
Reviewing now! 😊

**Review by ashleemboyer — APPROVED** [review]
Looks great! 🙌🏻 :shipit:

**Comment by zchsh** [comment]
Thanks for the reviews all 🙇 @BRKalow 🙌 Great callout, I've opened a PR to follow-up and fix that: #713

**Pull request merged**